### PR TITLE
UAR-462: Change logic for existing BOs

### PR DESCRIPTION
**Pull request by azafarCH**
### JIRA link
https://companieshouse.atlassian.net/jira/software/c/projects/UAR/boards/481?modal=detail&selectedIssue=UAR-462&sprint=2867

### Change description
Editing change link logic for existing BOs

### Work checklist

- [x] Tests added where applicable
- [x] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: test/controllers/check.your.answers.controller.spec.ts [hunk 1]
```diff
@@ -286,7 +286,6 @@ describe("GET tests", () => {
     expect(resp.text).toContain(`${BENEFICIAL_OWNER_OTHER_URL}${BO_OTHER_ID_URL}${CHANGE_LINKS.CHANGE_PRINCIPAL_ADDRESS}`);
     expect(resp.text).toContain(`${BENEFICIAL_OWNER_OTHER_URL}${BO_OTHER_ID_URL}${CHANGE_LINKS.LEGAL_FORM}`);
     expect(resp.text).toContain(`${BENEFICIAL_OWNER_OTHER_URL}${BO_OTHER_ID_URL}${CHANGE_LINKS.LAW_GOVERNED}`);
-    expect(resp.text).toContain(`${BENEFICIAL_OWNER_OTHER_URL}${BO_OTHER_ID_URL}${CHANGE_LINKS.START_DATE}`);
```
Comment (AndreKramer7654) [inline test/controllers/check.your.answers.controller.spec.ts:289]: why are we removing start date?

File: views/includes/check-your-answers/beneficial-owner-gov.html [hunk 2]
```diff
@@ -37,6 +37,16 @@
   {% endif %}
 {% endset %}
 
+{% if hiddenChangeLink == true %}
+  {% set changeLink = undefined %}
+{% else %}
+  {% set changeLink = CREATE_CHANGE_LINK(
+            changeLinkUrl + "/" + boc.id + OE_CONFIGS.START_DATE,
+            "Corporate beneficial owner " + boc.name + " - date it became a beneficial owner",
```
Comment (AndreKramer7654) [inline views/includes/check-your-answers/beneficial-owner-gov.html:45]: strange - it's boc here and not bog as below

File: views/includes/check-your-answers/beneficial-owner-individual.html [hunk 3]
```diff
@@ -85,7 +85,19 @@
...
+{% if hiddenChangeLink == true %}
+{% set rows = (rows.push({
+  key: {
+    text: "Date of birth"
+  },
+  value: {
+    text: boiDateOfBirth
+  },
+  actions: {
+  items: [undefined]
```
Comment (AndreKramer7654) [inline views/includes/check-your-answers/beneficial-owner-individual.html:97]: can we just not have actions or action.items?

File: views/update/update-check-your-answers.html [hunk 4]
```diff
@@ -21,6 +21,7 @@ <h1 class="govuk-heading-xl">Check your answers before sending your application
 
         {% set boTextDisplayed = false %}
         {% if (appData.beneficial_owners_individual and appData.beneficial_owners_individual.length > 0) %}
+          {% set hiddenChangeLink = true %}
```
Comment (AndreKramer7654) [inline views/update/update-check-your-answers.html:24]: why are we hiding change links for individual bos?

File: views/includes/check-your-answers/beneficial-owner-individual.html (hunk 3 above)
Comment (azafarCH) [reply views/includes/check-your-answers/beneficial-owner-individual.html:97]: tested them with Jitin, anything less than this seems to make the bottom border not cover the full width unfortunately

File: views/update/update-check-your-answers.html [hunk 5]
```diff
@@ -21,6 +21,7 @@ <h1 class="govuk-heading-xl">Check your answers before sending your application
...
           {% set existingIndividualBO = appData.beneficial_owners_individual | selectattr("ch_reference", "undefined") %}
```
[the remark this answers is not in the document]
Comment (azafarCH) [reply views/update/update-check-your-answers.html:25]: selectattr selects values that are not the second value, weirdly named, so if there is a value that != undefined, we'll get the existingBOs

File: views/includes/check-your-answers/beneficial-owner-gov.html (hunk 2 above)
Comment (azafarCH) [reply views/includes/check-your-answers/beneficial-owner-gov.html:45]: yeah, silly mistake from me, will also revert the what I did to start date in the test controller, thanks!

**Review by mmcpolin-ch — APPROVED** [review]

**Review by jkayyala — APPROVED** [review]

**Pull request merged**